src/modules/hip/kernel/flip_voxel.hpp

```diff
@@ -0,0 +1,188 @@
+#include <hip/hip_runtime.h>
+#include "rpp_hip_common.hpp"
+
+template <typename T>
+__global__ void flip_ncdhw_tensor(T *srcPtr,
+                                  uint3 srcStridesCDH,
+                                  T *dstPtr,
+                                  uint3 dstStridesCDH,
+                                  int channels,
+                                  uint3 mirrorXYZ,
+                                  RpptROI3DPtr roiGenericPtrSrc,
+                                  int batchIndex)
+{
+    int id_x = (hipBlockIdx_x * hipBlockDim_x + hipThreadIdx_x) * 8;        // W - inner most dim vectorized
+    int id_y = hipBlockIdx_y * hipBlockDim_y + hipThreadIdx_y;              // H - second to inner
+    int id_z = hipBlockIdx_z * hipBlockDim_z + hipThreadIdx_z;              // D - outer most dim
+
+    if ((id_z >= roiGenericPtrSrc->xyzwhdROI.roiDepth) || (id_y >= roiGenericPtrSrc->xyzwhdROI.roiHeight) || (id_x >= roiGenericPtrSrc->xyzwhdROI.roiWidth))
+    {
+        return;
+    }
+
+    int dstIdx = (id_z * dstStridesCDH.y) + (id_y * dstStridesCDH.z) + id_x;
+    int zFactor = ((mirrorXYZ.z) ? (roiGenericPtrSrc->xyzwhdROI.roiDepth - 1 - id_z) : (id_z + roiGenericPtrSrc->xyzwhdROI.xyz.z)) * srcStridesCDH.y;
+    int yFactor = ((mirrorXYZ.y) ? (roiGenericPtrSrc->xyzwhdROI.roiHeight - 1 - id_y) : (id_y + roiGenericPtrSrc->xyzwhdROI.xyz.y)) * srcStridesCDH.z;
+    int xFactor = id_x + roiGenericPtrSrc->xyzwhdROI.xyz.x;
+    if (mirrorXYZ.x)
+    {
+        // To handle the case when trying to load from invalid memory location when width is not a multiple of 8
+        if((batchIndex == 0) && (id_x + 8 > roiGenericPtrSrc->xyzwhdROI.roiWidth))
```

Comment:
why the kernel need to differentiate batchIndex? I think we need to do a code review for this PR

Reply:
For voxel kernels, we are launching each sample in a batch in a separate kernel
https://github.com/r-abishek/rpp/blob/ar/flip_voxel/src/modules/hip/kernel/flip_voxel.hpp#L147

This differentiation is needed when horizontal Flag is set to 1
in the kernel, we are loading as 8 pixel load and store and we are trying to load input in reverse manner when  horizontal Flag is set to 1

for output location 0, we are trying load 8 values from input (width - 1 - 8)th location. When the output width is a not a multiple of 8. The above formulae will result in negative source location when output location is  (width - 1 - 8). So to handle that case, we have added a separate condition for batchIndex = 0